R/generate_retrospective_data.R

```diff
@@ -0,0 +1,99 @@
+#' Generate retrospective reporting triangles
+#'
+#' This function ingests a reporting triangle and the number of retrospective
+#'   reporting triangles we want to create, and iteratively generates the
+#'   reporting triangle that would have been available as of the maximum
+#'   reference time, and iteratively working backwards for
+#'   `n_history_uncertainty` snapshots
+#'
+#' @param triangle_for_retro Matrix of the incomplete reporting triangle
+#'   to be used to generate retrospective nowcasts, with rows representing the
+#'   time points of reference and columns representing the delays
+#' @param n_history_uncertainty Integer indicating the number of retrospective
+#'   reporting triangles to be generated, always starting from the most
+#'   recent reference time. The default is to generate as many retrospective
+#'   reporting triangles that would have a sufficient number of rows to
+#'   estimate  a delay distribution and generate a nowcast from.
+#' @param n_history_delay Integer indicating the number of reference dates to be
+#'   used in the estimate of the reporting delay, always starting from the most
+#'   recent reporting delay. The default here is to us the difference between
```

Comment:
I would have thought a more standard default would be to default to this equalling the number of columns (so this ends up as being the last D reports).